File: Partition Array by Odd and Even.cpp
class Solution {
public:
    /**
     * @param nums: a vector of integers
     * @return: nothing
     */
    void partitionArray(vector<int> &nums) {
        // write your code here
        int left  = 0;
        int right = nums.size() - 1;
        while (left < right)
        {
            while (left < right && nums[left] % 2 == 1)
            {
                ++left;
            }
            
            if (left == right)
            {
                return;
            }
            
            while(left < right && nums[right] % 2 == 0)
            {
                --right;
            }
            
            if (left == right)
            {
                return;
            }
            
            swap(nums[left], nums[right]);
            ++left;
            --right;
        }
    }
};

File: Partition Array by Odd and Even.py
class Solution:
    # @param nums: a list of integers
    # @return: nothing
    def partitionArray(self, nums):
        # write your code here
        
        if not nums:
            return
        length = len(nums)
        left, right = 0, length - 1
        pivot = 0
        while left <= right:
            if nums[left] % 2 == 1:
                nums[pivot], nums[left] = nums[left], nums[pivot]
                pivot += 1
                left += 1
            else:
                left += 1
        
